tools/raid_xmltolua.cpp

```diff
@@ -0,0 +1,479 @@
+			std::cout << "[Error - parseSpawn] " << err.what() << std::endl;
+		}
+	}
+
+	return spawnList;
+}
+
+struct AreaSpawnEvent : public RaidEvent
+{
+	AreaSpawnEvent(uint32_t delay, Position from, Position to, uint32_t radius, std::vector<Spawn> spawnList) :
+			RaidEvent{delay}, from_{std::move(from)}, to{std::move(to)}, radius{radius}, spawnList{std::move(spawnList)} {}
+
+	static AreaSpawnEvent* from(uint32_t delay, const pugi::xml_node& eventNode, uint16_t radius)
+	{
+		if (radius == 0) {
+			Position from{
+				require_attr<int32_t>(eventNode, "fromx"),
+				require_attr<int32_t>(eventNode, "fromy"),
+				require_attr<int16_t>(eventNode, "fromz"),
+			};
+
+			Position to{
+				require_attr<int32_t>(eventNode, "tox"),
+				require_attr<int32_t>(eventNode, "toy"),
+				require_attr<int16_t>(eventNode, "toz"),
+			};
+
+			return new AreaSpawnEvent{delay, from, to, 0, parseSpawnList(eventNode)};
+		} else {
+			Position center{
+				require_attr<int32_t>(eventNode, "centerx"),
+				require_attr<int32_t>(eventNode, "centery"),
+				require_attr<int16_t>(eventNode, "centerz"),
+			};
+
+			return new AreaSpawnEvent{delay, {}, center, radius, parseSpawnList(eventNode)};
+		}
+	}
+
+	void to_script(std::ostringstream& ss) const override
+	{
+		if (radius == 0) {
+			ss << INDENT << "local from, to, z = {x=" << from_.x << ", y="
+				<< from_.y << "}, {x=" << to.x << ", y=" << to.y << "}, "
+				<< to.z << '\n';
+			for (auto&& spawn: spawnList) {
+				if (spawn.minAmount == spawn.maxAmount) {
+					ss << INDENT << "for _ = 1, " << spawn.minAmount
+					   << " do\n";
+				} else {
+					ss << INDENT << "for _ = 1, math.random("
+					   << spawn.minAmount << ", " << spawn.maxAmount
+					   << ") do\n";
+				}
+				ss << INDENT << INDENT
+				   << "local x, y = "
+				      "math.random(from.x, to.x), "
+				      "math.random(from.y, to.y)\n"
+				   << INDENT << INDENT << "Game.createMonster(\""
+				   << spawn.name << "\", Position(x, y, z))\n"
+				   << INDENT << "end\n";
+			}
+		} else {
+			ss << INDENT << "local center, radius, z = {x=" << to.x	<< ", y="
+				<< to.y << "}, " << radius << ", " << to.z << '\n';
+			for (auto&& spawn: spawnList) {
+				if (spawn.minAmount == spawn.maxAmount) {
+					ss << INDENT << "for _ = 1, " << spawn.minAmount
+					   << " do\n";
+				} else {
+					ss << INDENT << "for _ = 1, math.random("
+					   << spawn.minAmount << ", " << spawn.maxAmount
+					   << ") do\n";
+				}
+				ss << INDENT << INDENT
+				   << "local x, y = "
+				      "math.random(center.x - radius, center.x + radius), "
+				      "math.random(center.y - radius, center.y + radius)\n"
+				   << INDENT << INDENT << "Game.createMonster(\""
+				   << spawn.name << "\", Position(x, y, z))\n"
+				   << INDENT << "end\n";
+			}
+		}
+	}
+
+	Position from_, to;
+	uint32_t radius;
+	std::vector<Spawn> spawnList;
+};
+
+struct ScriptEvent : public RaidEvent
+{
+	ScriptEvent(uint32_t delay) : RaidEvent{delay} {}
```

Comment:
Not a big issue, but I kinda dislike storing quantities that have "units" (in this case, time) in uint32_t.

Reply:
What do you suggest?

Reply:
`uint32_t` => `std::chrono::milliseconds` (or w/e unit it uses ATM).